core/src/main/java/org/elasticsearch/index/query/MatchQueryBuilder.java

```diff
@@ -60,8 +60,10 @@
     public static final ParseField TYPE_FIELD = new ParseField("type");
     public static final ParseField QUERY_FIELD = new ParseField("query");
 
-    /** The default name for the match query */
-    public static final String NAME = "match";
+    /** The names for the match query. */
+    // TODO replace this with a Types like object
+    public static final Names NAMES = new Names("match", "match_phrase", "matchPhrase", "match_phrase_prefix", "matchPhrasePrefix",
```

Comment:
I agree with @nik9000 here. While ParseFields could be reused here, that also slightly twisting their semantics, since the whole notion of deprecation does not always apply here. 
Also we'd the names mixed with the whole CamelCase magix in ParseFields, which I personally would try to avoid, and the "primary" name that we use as identifier on the NamedWriteableAwareStreamInput is something slightly different from the "preffered" name in the ParseFields. Its mostly naming, but I think here the distinction should be clear.